### PR TITLE
Problem: testground testcase output artifact not collectiable in k8s workflow

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- create static filename for argo artifact collection

The k8s workflow will have a static path to collect the artifacts: `/data.tar.bz2`

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **New Features**
  - Improved output data handling by creating tarballs in a temporary location before transferring to the final output directory.

- **Bug Fixes**
  - Enhanced error reduction related to file handling during tarball creation and output.

- **Refactor**
  - Simplified the `run` function by removing unnecessary filename generation logic, leading to clearer code structure.

### CODE REVIEW

File: testground/benchmark/benchmark/stateless.py [hunk 1]
```diff
@@ -146,12 +146,14 @@ def run(
         proc.wait()
 
         # collect outputs
+        output = Path("/data.tar.bz2")
+        with tarfile.open(output, "x:bz2") as tar:
+            tar.add(home, arcname="data")
         outdir = Path(outdir)
         if outdir.exists():
```
Comment (mmsqe) [inline testground/benchmark/benchmark/stateless.py:153]: should we check like is_dir?

Comment (yihuang) [reply testground/benchmark/benchmark/stateless.py:153]: I think it's fine, it'll raise error if the mount is not a directory.

Comment (mmsqe) [reply testground/benchmark/benchmark/stateless.py:153]: I see, but just get `IsADirectoryError: [Errno 21] Is a directory: '/outputs'` in `filename.unlink(missing_ok=True)`

Comment (yihuang) [reply testground/benchmark/benchmark/stateless.py:153]: added an specific assert

**Review by mmsqe — APPROVED** [review]

**Pull request merged**